routes/index.js

```diff
@@ -35,23 +35,25 @@ exports.index = function (req, res, next) {
 
 
 exports.admin = function (req, res, next) {
-  console.log(req.body);
-  User.find({ username: req.body.username, password: req.body.password }, function (err, users) {
-    if (users.length > 0) {
-      return res.render('admin', {
-        title: 'Admin Access Granted',
-        granted: true,
-      });
-    } else {
-      return res.render('admin', {
-        title: 'Admin Access',
-        granted: false,
-      });
+    const { username, password } = req.body;
+    const user = getUserByUsername(username);
+
+    if (!user || user.password == password) {
+        return res.status(401).send('Unauthorized');
     }
-  });
+
+    return res.status(200).send('Login successful');
 
 };
 
+function checkAndIncreaseLoginCount(user: User): User {
+  if (user.loginCount > 0) {
+    user.loginCount += 1;
+  }
+  return user;
+}
```

Comment:
The function is mutating the original user object, which can lead to unintended side effects, especially if the function is used in multiple places or if immutability is preferred for state management.

```suggestion
function checkAndIncreaseLoginCount(user: User): User {
  return {
    ...user,
    loginCount: user.loginCount + 1
  };
}
```